src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -727,6 +726,42 @@ export class EditorComponent implements OnInit {
     this.disableSubmit = this.disableFloodFill = false;
   }
 
+  getFloodfillSet(coord: Coordinate) {
+    const maskSet = this.magicWandService.floodfill(
+      this.originalImageData,
+      coord.x,
+      coord.y, 
+      this.tolerance
+    )
+    this.floodfillMask(
+      new MaskAction(          
+        this.maskTool == MaskTool.MAGIC_WAND_ADD
+            ? Action.ADD
+            : Action.SUBTRACT,
+          Tool.MAGIC_WAND,
+          maskSet
+    ));
+  }
+
+  async getScribbleSet(coord: Coordinate): Promise<void> {
+    const paintedSet = await this.getPaintedSet();
+    const maskSet = this.magicWandService.scribbleFloodfill (
+      this.originalImageData,
+      coord.x,
+      coord.y, 
+      this.tolerance,
+      paintedSet
+    )
+    this.floodfillMask(
+      new MaskAction(
+        this.maskTool == MaskTool.MAGIC_WAND_ADD
+          ? Action.ADD
+          : Action.SUBTRACT,
+        Tool.SCRIBBLE,
+        maskSet
+    ));
+  }
+
```

Comment:
Hmm, am I correct in understanding that you're trying to merge a lot of what mask-directive handles into editor component? I don't know how merging is gonna work, because I implemented my Preview-Floodfills algo by branching on the current version of master, and that version still uses mask-directive to handle a lot of magicWandService...that means we're gonna have to re-implement my preview-floodfills algo on the UI side, which is gonna require both of us at this point, because you won't be up-to-date with the functions and code that I added, and I may not understand all the implications of the changes that you've made here too. Something to keep in mind for Tues 1|Sept|2020...